plugins/destiny.py

```diff
@@ -29,30 +34,34 @@ def strip_tags(html):
     s.feed(html)
     return s.get_data().replace('\n','\t')
 
-def get_membership(user_name, bot):
+def get_membership(user_name):
     """
     Takes in a username and returns a dictionary of all systems they are
     on as well as their associated id for that system
     """
+    user_name = CACHE['links'].get(user_name, user_name)
     if CACHE.get(user_name, None):
         return CACHE[user_name]
     else:
-        userID = get(
-            "http://www.bungie.net/Platform/User/SearchUsers/?q={}"
-            .format(user_name.strip()),
-            headers=HEADERS).json()['Response'][0]
-        userIDHash = userID['membershipId']
-        userName = userID['displayName']
-
-        userHash = get(
+        try:
+            userID = get(
+                "http://www.bungie.net/Platform/User/SearchUsers/?q={}".format(user_name.strip()),
+                headers=HEADERS).json()['Response'][0]
+            userIDHash = userID['membershipId']
+            userName = userID['displayName']
+
+            userHash = get(
             "https://www.bungie.net/platform/User/GetBungieAccount/{}/254/"
             .format(userIDHash),
             headers=HEADERS).json()['Response']['destinyAccounts']
+        except:
+            return "A user by the name {} was not found.".format(user_name)
+
         membership = {}  # {console: memberId}
         for result in userHash:
             membership[result['userInfo']['membershipType']] = result['userInfo']['membershipId']
         CACHE[userName] = membership
-        return membership
+        return membership if membership != {} else "A user by the name {} was not found.".format(user_name)
```

Comment:
sometimes a member can be found above but not have anything in the Destiny database. possibly accounts from the Halo era?